### PR TITLE
Fixes issues where autoupdate ignores mysql.columnName and deletes columns

### PR DESCRIPTION
**Pull request by rlloyd2001**
Fixes these issues:
https://github.com/strongloop/loopback-connector-mysql/issues/168
https://github.com/strongloop/loopback/issues/2085

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by rlloyd2001** [comment]
so it looks like my 'commit' notes are to lengthy on the same line, I will update these commits and try again

File: test/migration.test.js [hunk 1]
```diff
@@ -338,6 +338,30 @@ describe('migrations', function() {
...
+    db.autoupdate('ColRenameTest', function (err) {
+      db.discoverModelProperties('col_rename_test', function (err, props) {
+        assert.equal(props[0].columnName, 'first_name');
+        assert.equal(props[1].columnName, 'lastName');
+        assert.equal(props.length, 2);
+        done();
+      });
+    });
+  });
+
```
Comment (rlloyd2001) [inline test/migration.test.js]: Can remove redundancy and I think this test can be moved to the mysql.autoupdate.test.js file

File: test/mysql.autoupdate.test.js [hunk 2]
```diff
@@ -176,4 +211,25 @@ describe('MySQL connector', function() {
       done();
     });
   });
+
+  it('"mysql.columnName" is updated with correct name on create table', function (done) {
```
Comment (superkhau) [inline test/mysql.autoupdate.test.js]: remove space after function

DId you run the linter?

File: test/mysql.autoupdate.test.js [hunk 3]
```diff
@@ -176,4 +211,25 @@ describe('MySQL connector', function() {
...
+    // first autoupdate call uses create table
+    verifyMysqlColumnNameAutoupdate(done);
+  });
+
+  it('"mysql.columnName" is updated without changing column name on alter table', function (done) {
```
Comment (superkhau) [inline test/mysql.autoupdate.test.js]: ditto

**Comment by superkhau** [comment]
LGTM

@raymondfeng Can you PTAL too?

**Comment by superkhau** [comment]
@slnode test please

File: test/mysql.autoupdate.test.js (hunk 2 above)
Comment (rlloyd2001) [inline test/mysql.autoupdate.test.js]: Thanks, first PR attempt on this framework. I did not run the linter. Are there instructions on running the linter for this framework I missed?

Comment (superkhau) [inline test/mysql.autoupdate.test.js]: @rlloyd2001 Thanks for the contribution, I'll try to help you out on the way. The linter runs automatically when you run `npm test` or you can run it manually via `npm run lint` -- see https://github.com/strongloop/loopback-connector-mysql/blob/master/package.json#L8.

**Comment by rlloyd2001** [comment]
lint errors should be fixed now

**Comment by superkhau** [comment]
@slnode test please

File: test/mysql.autoupdate.test.js [hunk 4]
```diff
@@ -163,6 +167,37 @@ describe('MySQL connector', function() {
...
+            'dataLength': 40,
+          },
+        },
+        'lastName': {
+          'type': 'String',
+          'required': false,
+          'length': 40,
+        },
+      },
+    };
```
Comment (superkhau) [inline test/mysql.autoupdate.test.js]: Remove the single quotes around all keys.

**Comment by rlloyd2001** [comment]
@superkhau  The styling change is in.

**Comment by superkhau** [comment]
@slnode test please

**Comment by rlloyd2001** [comment]
@superkhau  It looks like the tests that are failing are not related to this pull request update. I also noticed 6 fails when I ran locally on master, but I think they might be from having a different MySql version.

**Comment by rlloyd2001** [comment]
What's the next steps to get this merged? Do all the tests need to be green? It looks like they are def. failing from external factors not from this PR. I'd be happy to help fix them, but I might need to be pointed in the right direction or more server access.

**Comment by superkhau** [comment]
@rlloyd2001 No the tests do not all need to be green. Just the related ones need to be. Unfortunately, our CI doesn't allow external access yet (I've been pushing to for this to happen, but it'll take awhile till we work that part out). Anyways, I'll have to review and dig a bit more into the failures. Will ping you again soon.

**Comment by superkhau** [comment]
LGTM from @raymondfeng on Slack.

**Comment by superkhau** [comment]
@rlloyd2001 Once I confirm all CI issues are unrelated, we can get this merged.

**Comment by rlloyd2001** [comment]
This is needed so we can configure indexes in tables using loopback without the rows being deleted, so any help getting this merged would be appreciated. We can configure indexes manually or using the connector, but I was hoping to use loopback for the indexes.

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by superkhau** [comment]
test please

**Comment by superkhau** [comment]
@rlloyd2001 Can you rebase the PR? It's out of date now. TY

**Comment by rlloyd2001** [comment]
@superkhau rebased

**Comment by superkhau** [comment]
test please

**Comment by superkhau** [comment]
Thanks for the contribution! :+1: Merging as the failures are unrelated to this PR.

**Pull request merged**